app/controllers/product_drives_controller.rb

```diff
@@ -85,6 +94,13 @@ def destroy
 
   private
 
+  def verify_role
```

Comment:
Can we move this verification to the destroy service? You can pass the user into it.